amp/WEB-INF/src/org/dgfoundation/amp/visibility/data/FMTree.java

```diff
@@ -89,20 +89,23 @@ public FMTree(Map<String, FMTree> entries, boolean enabled) {
      *  }
      * </pre>
      */
-    public Map<String, Object> asMap(boolean fullEnabledPaths) {
-        Map<String, Object> mapTree = new HashMap<>();
-        if (!fullEnabledPaths || this.enabled)
+    public FMSettingsTree asFmSettingsTree(boolean fullEnabledPaths) {
+        FMSettingsTree fmSettingsTree = new FMSettingsTree();
+        if (!fullEnabledPaths || this.enabled) {
             if (!fullEnabledPaths) {
-                mapTree.put(EPConstants.FM_ENABLED, this.enabled);
+                fmSettingsTree.setEnabled(this.enabled);
```

Comment:
Oh, I have just noticed that the existing `FMTree` matches `FMSettingsTree` (funny to notice the consistency over time :D), so may be just to use it directly instead?